R/lint.R

```diff
@@ -85,7 +85,12 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
   if (!is_tainted(source_expressions$lines)) {
     for (expr in source_expressions$expressions) {
       for (linter in names(linters)) {
-        lints[[length(lints) + 1L]] <- get_lints(expr, linter, linters[[linter]], lint_cache, source_expressions$lines)
+        lints[[length(lints) + 1L]] <- withCallingHandlers(
+          get_lints(expr, linter, linters[[linter]], lint_cache, source_expressions$lines),
+          error = function(cond) {
+            stop("Linter ", linter, "() failed in ", filename, ": ", conditionMessage(cond), call. = FALSE)
```

Comment:
Would drop the (), because custom linter naming would cause misleading error messages.

Reply:
Good point, thanks